src/db.c

```diff
@@ -257,6 +257,7 @@ void setKey(client *c, redisDb *db, robj *key, robj *val, int flags) {
 
     if (!keyfound) {
         dbAdd(db,key,val);
+        notifyKeyspaceEvent(NOTIFY_NEW,"new",key,db->id);
```

Comment:
i think this should go into dbAdd (many places call it directly)

Reply:
Moved, thanks for looking. Please let me know if there are tests you'd like to see to cover this.

No problem on the version, I can emulate this with Lua temporarily.